packages/popper/src/index.vue

```diff
@@ -0,0 +1,466 @@
+<template>
+  <slot name="trigger"></slot>
+  <transition :name="transition">
+    <div
+      v-show="visible"
+      :id="popperId"
+      ref="popperRef"
+      role="tooltip"
+      :aria-hidden="visible ? 'false' : 'true'"
+      :class="['el-popper', 'is-' + effect, popperClass]"
+      @mouseenter="_show"
+      @mouseleave="_hide"
+    >
+      <!-- mark this parent container un-indexable to disable it from being indexed -->
+      <slot>
+        <div>{{ content }}</div>
+      </slot>
+      <div
+        v-if="showArrow"
+        ref="arrowRef"
+        class="el-popper__arrow"
+        data-popper-arrow
+      ></div>
+    </div>
+  </transition>
+</template>
+
+<script lang="ts">
+import {
+  computed,
+  defineComponent,
+  getCurrentInstance,
+  ref,
+  onBeforeUnmount,
+  onMounted,
+  onUpdated,
+  watch,
+  Fragment,
+} from 'vue'
+import { isArray } from '@vue/shared'
+import { debounce } from 'lodash'
+import { createPopper } from '@popperjs/core'
+
+import { generateId } from '@element-plus/utils/util'
+import { on, off } from '@element-plus/utils/dom'
+
+import useModifer from './useModifier'
+
+import type { PropType, Ref } from 'vue'
+import type {
+  Placement,
+  Instance as PopperInstance,
+  PositioningStrategy,
+} from '@popperjs/core'
+import useModifier from './useModifier'
+
+type Effect = 'dark' | 'light';
+
+const stop = (e) => e.stopPropagation()
+const throwError = (e: string) => {
```

Comment:
can we put this function into utils?

Reply:
Maybe we can make it a custom error for ElementPlus